libursa/OnDiskDataset.cpp

```diff
@@ -294,9 +307,6 @@ const OnDiskIndex &OnDiskDataset::get_index_with_type(
 }
 
 void OnDiskDataset::drop_file(const std::string &fname) const {
-    // it may happen that dataset was reloaded and then is scheduled for removal
-    // multiple times so we have to account for that and only delete yet
-    // existing files
```

Comment:
is this no longer relevant after other refactors?

Reply:
Actually that's one of the reasons why I've added you here. I don't understand that comment, so I *think* it's no longer relevant.

1) dataset shouldn't be removed multiple times, at least not with the current code version. Not 100% sure but I think

2) fs::remove won't crash if the file did not exist (it'll just return false).

3. despite the comment, the code doesn't check if the file exists or not :thinking:

Reply:
I've checked that this comment is with us since initial public release (2 years ago), also:

* I don't remember why exactly it was placed
* Seems like it's actually no longer relevant for numerous reasons